pkg/controller/ticdc_control.go

```diff
@@ -63,17 +94,193 @@ func (c *defaultTiCDCControl) GetStatus(tc *v1alpha1.TidbCluster, ordinal int32)
 	return &status, err
 }
 
+func (c *defaultTiCDCControl) DrainCapture(tc *v1alpha1.TidbCluster, ordinal int32) (int, bool, error) {
+	httpClient, err := c.getHTTPClient(tc)
+	if err != nil {
+		klog.Warningf("ticdc control: drain capture is failed, error: %v", err)
+		return 0, false, err
+	}
+
+	baseURL := c.getBaseURL(tc, ordinal)
+
+	captures, retry, err := getCaptures(httpClient, baseURL)
+	if err != nil {
+		klog.Warningf("ticdc control: drain capture is failed, error: %v", err)
+		return 0, false, err
+	}
+	if retry {
+		// Let caller retry drain capture.
+		return 0, true, nil
+	}
+	if len(captures) == 1 {
+		// No way to drain a single node TiCDC cluster, ignore.
+		return 0, false, nil
+	}
+
+	this, owner := getOrdinalAndOwnerCaptureInfo(tc, ordinal, captures)
+	if this == nil {
+		addr := getCaptureAdvertiseAddressPrefix(tc, ordinal)
+		return 0, false, fmt.Errorf("capture not found, address: %s, captures: %+v", addr, captures)
+	}
+	if owner == nil {
+		return 0, false, fmt.Errorf("owner not found, captures: %+v", captures)
+	}
+
+	payload := drainCaptureRequest{
+		CaptureID: this.ID,
+	}
+	payloadBody, err := json.Marshal(payload)
+	if err != nil {
+		return 0, false, fmt.Errorf("ticdc drain capture failed, marshal request error: %v", err)
+	}
+	req, err := http.NewRequest("PUT", baseURL+"/api/v1/captures/drain", bytes.NewReader(payloadBody))
+	if err != nil {
+		return 0, false, fmt.Errorf("ticdc drain capture failed, new request error: %v", err)
+	}
+	res, err := httpClient.Do(req)
+	if err != nil {
+		return 0, false, fmt.Errorf("ticdc drain capture failed, request error: %v", err)
+	}
+	defer httputil.DeferClose(res.Body)
+	if res.StatusCode == http.StatusNotFound {
+		// It is likely the TiCDC does not support the API, ignore.
+		klog.Infof("ticdc control: %s does not support drain capture, skip", this.AdvertiseAddr)
+		return 0, false, nil
+	}
+	if res.StatusCode == http.StatusServiceUnavailable {
+		// TiCDC is not ready, retry.
+		klog.Infof("ticdc control: %s service unavailable drain capture, retry", this.AdvertiseAddr)
+		return 0, true, nil
+	}
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return 0, false, fmt.Errorf("ticdc drain capture failed, read response error: %v", err)
+	}
+
+	var resp drainCaptureResp
+	err = json.Unmarshal(body, &resp)
+	if err != nil {
+		// It is likely the TiCDC does not support the API, ignore.
```

Comment:
Is this correct for the unmarshal error?

Reply:
Yes, we ignore the error because old ticdc does not support this API.